src/storage/v2/disk/storage.cpp

```diff
@@ -376,9 +376,11 @@ std::optional<EdgeAccessor> DiskStorage::DiskAccessor::DeserializeEdge(const roc
   }
   const auto edge_type_id = storage::EdgeTypeId::FromUint(std::stoull(edge_parts[3]));
   auto maybe_edge = CreateEdge(&*from_acc, &*to_acc, edge_type_id, edge_gid, value.ToStringView(), key.ToString());
-  MG_ASSERT(maybe_edge.HasValue());
-
-  return *maybe_edge;
+  if (maybe_edge.HasValue()) {
+    return *maybe_edge;
+  }
+  // One of vertex for edge could be deleted, so we don't create this edge.
+  return std::nullopt;
```

Comment:
Can you also explain what is happening here? Because you were saying to me that `DeserializeEdge` is a problem because edge can be deleted. But `PrefetchEdges` shouldn't prefetch deleted edges! And if that is the test when properties on edges are false I think I know where the bug comes from:
```
      if (config_.properties_on_edges) {
        if (auto maybe_old_disk_key = utils::GetOldDiskKeyOrNull(edge.ptr->delta); maybe_old_disk_key.has_value()) {
          if (!DeleteEdgeFromDisk(maybe_old_disk_key.value())) {
            return StorageDataManipulationError{SerializationError{}};
          }
        }
      }
```
We only delete edge if properties on edges are true.

Reply:
The main issue is that we deleted edge in the same transaction, so changes to disk were not written yet.
In prefetch edges we read edges from disk and the deleted edge is still presented on disk and can be prefetched. That's the reason why we need to filter out such edges during edge prefetch.